### PR TITLE
feat: `HEAD`, `OPTIONS`, `PATCH`, `TRACE` request methods

### PR DESCRIPTION
**Pull request by fabrice8**
## Description

1. I noticed the router interface doesn't provide `HEAD`, `OPTIONS`, `PATCH`, `TRACE` request methods yet.
2. And I proposed changing `res.set` to `res.header` in the response class, to make it explicit that headers are those being set.
2. I also added a group of test to test some of the methods added 

Kindly check changes in the PR codes, they might be more self-explanatory. And fill free to change the code at your convenience.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by codekeyz** [comment]
This is good stuff @fabrice8. I'll look into this.

File: lib/src/http/response.dart [hunk 1]
```diff
@@ -217,8 +225,8 @@ class Response extends Message<shelf.Body?> implements $Response {
         headers: headers,
         ended: ended,
       )
-          .status(500)
-          .json(makeError(message: message ?? 'Internal Server Error').toJson);
+          ..status(500)
+          ..json(makeError(message: message ?? 'Internal Server Error').toJson);
```
Comment (codekeyz) [inline lib/src/http/response.dart:229]: Why this change?

File: pharaoh_examples/lib/api_service/index.dart [hunk 2]
```diff
@@ -55,6 +55,17 @@ void main() async {
...
+  /// example: http://localhost:3000/api/users
+  app.head('/api/*', (req, res) => res.status(200).send(''));
+
+  /// Patch request
+  app.patch('/api/user/:name/repos', (req, res) {
+    var name = req.params['name'];
+
+    return res.status(200).send('Updated $name repos');
+  });
+
```
Comment (codekeyz) [inline pharaoh_examples/lib/api_service/index.dart:68]: This shouldn't be here. When you add a new feature to Pharaoh, you put the tests in the `lib/test` directory.

File: pharaoh_examples/test/api_service_test.dart [hunk 3]
```diff
@@ -10,6 +10,26 @@ void main() async {
...
+
+      test('when using `PATCH` request method', () async {
+        final serverUrl = apisvc.app.uri.toString();
+        final path = Uri.parse('$serverUrl/api/user/jane/repos?api-key=foo');
+
+        final result = await http.patch(path);
+        expect(result.statusCode, 200);
+        expect(result.body, 'Updated jane repos');
+      });
+    });
```
Comment (codekeyz) [inline pharaoh_examples/test/api_service_test.dart:31]: This shouldn't be here.

**Review by codekeyz — CHANGES_REQUESTED** [review]
Amazing work on the PR, but we'll need to relook at a few things. Let me know if you do need any clarification.

File: pharaoh_examples/test/api_service_test.dart (hunk 3 above)
Comment (fabrice8) [reply pharaoh_examples/test/api_service_test.dart:31]: Moving this to a new file `pharaoh_examples/test/api_method_test.dart`

Comment (fabrice8) [reply pharaoh_examples/test/api_service_test.dart:31]: this isn't useful without [this](https://github.com/codekeyz/pharaoh/pull/45/files/c8f0cc0daf36b924dfe062e9823c5a93fd854934#diff-476633b5ed9ab7c51840c4ae6349c833d39fa57ff41b25565fe165b0499667f3) so dropped it

File: pharaoh_examples/lib/api_service/index.dart (hunk 2 above)
Comment (fabrice8) [reply pharaoh_examples/lib/api_service/index.dart:68]: Dropped this change as well

File: lib/src/http/response.dart (hunk 1 above)
Comment (fabrice8) [reply lib/src/http/response.dart:229]: Dropped this change
